engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
+     * object to allow it to perform its damage handling. This usually means just subtracting the damage from object's
+     * health and playing a hit sound, but if the object has no health pool or should be otherwise indestructible, or
+     * invincible against some types of damage, this method can be freely left blank.
+     *
+     * @param dmg      Damage the object receives.
+     * @param game     Game this object belongs to.
+     * @param position Position the object was hit at, if hit by point-based damage. Null if not applicable, such as fire.
+     * @param dmgType  Type of the damage object receives.
+     */
     void receiveDmg(float dmg, SolGame game, @Nullable Vector2 position, DmgType dmgType);
 
+    /**
+     * Denotes whether this object is affected by gravity, this is usually constant.
+     *
+     * @return True if object is affected by gravity, false otherwise.
+     */
     boolean receivesGravity();
 
+    /**
+     * Applies external force to object.
+     * <p>
+     * This method usually just scales the {@code force} by objects mass if {@code acc} is set, and then passes the
+     * force to object's internal {@link Body}.
+     *
+     * @param force Force to apply to the object.
+     * @param game  Game this object belongs to.
+     * @param acc   Whether this objects is accelerating by the force, such as in gravity.
+     */
     void receiveForce(Vector2 force, SolGame game, boolean acc);
 
+    /**
+     * Returns object's position.
+     * <p>
+     * This method usually just returns the vector retrieved from object's internal {@link Body}.
+     *
+     * @return Position of the object.
+     */
     Vector2 getPosition();
 
+    /**
+     * Called when an object is too far from player/camera, and is thus to be converted to {@link FarObject}.
+     * <p>
+     * This method is used for optimization of game's objects, by converting to their more resource friendly versions
+     * when they get too far from player/camera. The {@code FarObject} returned by this method should be usable to get
+     * copy of this object back in reverse. If null is returned by this method, this object is designated to be removed
+     * when it gets too far from player/camera, and instead of recreating from {@code FarObject}, to be generated again.
+     *
+     * @return FarObject representation of this object, or null if object is to be removed.
+     */
     FarObject toFarObject();
 
+    /**
+     * Returns list of all {@link Drawable Drawables} this object has assigned.
+     *
+     * @return List of drawables this object has.
+     */
     List<Drawable> getDrawables();
 
+    /**
+     * Returns object's angle.
+     * <p>
+     * This method usually just returns the angle retrieved from object's internal {@link Body}.
+     *
+     * @return Angle of this object, in degrees.
+     */
     float getAngle();
 
+    /**
+     * Returns object's speed.
```

Comment:
`* Return the object's speed.`

Side note: This method should be renamed to `getVelocity` as soon as possible.